From 34c17749aee2f3da9ac3cb984708e5fa88e9ed30 Mon Sep 17 00:00:00 2001
From: monkuous <84319635+monkuous@users.noreply.github.com>
Date: Fri, 16 May 2025 12:04:58 +0200
Subject: [PATCH] Add Hydrogen support

---
 m4/libtool.m4 | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 8d323b3..c58eada 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -3008,6 +3008,18 @@ netbsd*)
   hardcode_into_libs=yes
   ;;
 
+*-plibc)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  dynamic_linker='plibc ld.so'
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=yes
+  ;;
+
 newsos6)
   version_type=linux # correct to gnu/linux during the next big refactor
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
-- 
2.49.0

